src/lamp_py/ingestion/convert_gtfs_rt.py

```diff
@@ -397,33 +385,76 @@ def sync_with_s3(self, local_path: str) -> bool:
 
         return False
 
-    def merge_local_pq(self, table: pyarrow.Table, local_path: str) -> None:
+    def write_local_pq(self, table: pyarrow.Table, local_path: str) -> None:
         """
         merge pyarrow Table with existing local_path parquet file
 
         :param table: pyarrow Table
         :param local_path: path to local parquet file
         """
-        local_folder = local_path.replace(os.path.basename(local_path), "")
-        new_table_path = os.path.join(
-            local_folder,
-            "new_table.parquet",
+        part_col = self.detail.partition_column
+        table = hash_gtfs_rt_table(table)
+        out_ds = pd.dataset(table)
+
+        if self.sync_with_s3(local_path) is True:
+            hash_gtfs_rt_parquet(local_path)
+            out_ds = pd.dataset(
+                [
+                    pd.dataset(table),
+                    pd.dataset(local_path),
+                ]
+            )
+
+        partitions = pc.unique(
+            out_ds.to_table(columns=[part_col]).column(part_col)
         )
-        pq.write_table(table, new_table_path)
 
-        join_table_path = os.path.join(
-            local_folder,
-            "join_table.parquet",
+        with tempfile.TemporaryDirectory() as temp_dir:
+            tmp_pq = os.path.join(temp_dir, "temp.parquet")
+            with pq.ParquetWriter(tmp_pq, schema=out_ds.schema) as writer:
+                for part in partitions:
+                    part_filter = pc.field(part_col) == part
+                    table = pl.from_arrow(out_ds.to_table(filter=part_filter))
+                    writer.write_table(
+                        table.sort(
+                            by=["feed_timestamp"],
+                        )
+                        .unique(subset=GTFS_RT_HASH_COL, keep="first")
+                        .to_arrow()
+                        .cast(out_ds.schema)
```

Comment:
de-dupe operation happens here for each individual partition, doing this by partition will limit memory usage during de-duper operation.

There are most likely way to optimize/speed this up when parquet file sizes start to increase